front/src/monthview/Monthview.js

```diff
@@ -0,0 +1,479 @@
+              </div>
+            )}
+        </Link>,
+      );
+      firstMon.add(1, 'days');
+      info = false;
+    }
+
+    // lisätään kuukauteen kuuluvat päivät.
+    while (startDay.format('D') < days) {
+      const colorFromBackEnd = checkColor(this.state.paivat, help);
+      if (this.state.paivat[help].tracks) {
+        this.state.paivat[help].tracks.forEach((track) => {
+          if (track.notice !== null) {
+            info = true;
+          }
+        });
+      }
+      help += 1;
+      table.push(
+        <Link style={{ backgroundColor: `${colorFromBackEnd}` }} class="link" to={`/dayview/${startDay.format('YYYY-MM-DD')}`}>
+          {info
+            ? (
+              <div className="monthview-day-img">
+                {startDay.date()}
+                <img className="Monthview-info" src={smallInfoIcon} />
+              </div>
+            )
+            : (
+              <div>
+                {startDay.date()}
+              </div>
+            )}
+        </Link>,
+      );
+      startDay.add(1, 'days');
+      info = false;
+    }
+
+    let colorFromBackEnd = checkColor(this.state.paivat, help);
+    help += 1;
+    if (this.state.paivat[help].tracks) {
+      this.state.paivat[help].tracks.forEach((track) => {
+        if (track.notice !== null) {
+          info = true;
+        }
+      });
+    }
+    table.push(
+      <Link style={{ backgroundColor: `${colorFromBackEnd}` }} class="link" to={`/dayview/${startDay.format('YYYY-MM-DD')}`}>
+        {info
+          ? (
+            <div className="monthview-day-img">
+              {startDay.date()}
+              <img className="Monthview-info" src={smallInfoIcon} />
+            </div>
+          )
+          : (
+            <div>
+              {startDay.date()}
+            </div>
+          )}
+      </Link>,
+    );
+    startDay.add(1, 'days');
+
+    // Lisätään viimeiseen viikkoon kuuluvat päivät.
+    while (startDay.format('ddd') !== 'Mon' && startDay.format('ddd') !== 'ma') {
+      colorFromBackEnd = checkColor(this.state.paivat, help);
+      if (this.state.paivat[help].tracks) {
+        this.state.paivat[help].tracks.forEach((track) => {
+          if (track.notice !== null) {
+            info = true;
+          }
+        });
+      }
+      help += 1;
+      table.push(
+        <Link style={{ backgroundColor: `${colorFromBackEnd}` }} class="link notCurMonth" to={`/dayview/${startDay.format('YYYY-MM-DD')}`}>
+          {info
+            ? (
+              <div className="monthview-day-img">
+                {startDay.date()}
+                <img className="Monthview-info" src={smallInfoIcon} />
+              </div>
+            )
+            : (
+              <div>
+                {startDay.date()}
+              </div>
+            )}
+        </Link>,
+      );
+      startDay.add(1, 'days');
+      if (safetyLoop > 7) {
+        break;
+      }
+      safetyLoop += 1;
+      info = false;
+    }
+
+    return table;
+  }
```

Comment:
Looks like there's a bunch of repeating code, could this be shortened somehow?

Reply:
In theory the parts inside while loops could be made function that gets called since only the while conditions differ

Reply:
Only the while loops differ, so i made the inside code into a function that get's called by while loops. saved around 100 lines of code